src/layer/place.js

```diff
@@ -187,7 +181,31 @@ export const city = {
         [11, 0.9],
       ],
     },
-    "text-field": Label.localizedName,
+    "text-field": [
+      "let",
+      "localizedName",
+      "",
+      "localizedCollator",
+      ["collator", {}],
+      [
+        "case",
+        [
+          "==",
+          ["var", "localizedName"],
+          ["get", "name"],
+          ["var", "localizedCollator"],
+        ],
+        ["var", "localizedName"],
+        // Gloss the name in the local language if it differs from the localized name.
+        [
+          "format",
+          ["var", "localizedName"],
+          "\n",
+          ["concat", "(", ["get", "name"], ")"],
```

Comment:
I worked around this issue, but the workaround was not pretty. It relies on some garbled text that I’ve made invisible, but we’d need to revisit this approach if we ever enable screen reader support on this map.

[<img src="https://user-images.githubusercontent.com/1231218/204648807-6e6a5e7e-5668-4947-8e2b-54bcc6475378.png" width="800" alt="Tunis">](https://zelonewolf.github.io/openstreetmap-americana/#map=6.66/36.464/8.617&language=en)
[<img src="https://user-images.githubusercontent.com/1231218/204649039-52b06d2e-62c9-497e-bae2-1602aaddd2fd.png" width="800" alt="Seville">](https://zelonewolf.github.io/openstreetmap-americana/#map=8/37.327/-6.468&language=ar)

Reply:
Are the parentheses really necessary?

Reply:
I think the gloss does need something to set it apart, since either name can wrap onto multiple lines. If not parentheses, then the gloss would need to be much smaller or in a different font or color. But a smaller label might cause the user to mistake the city label for a village label that happens to be near a city label, and a lighter label might be mistaken for a state.

Reply:
Parentheses are the typical way to show that a different name is used in the local language as the example in #471 shows.  However this example is with two languages that both use the Roman alphabet.  Maybe when the writing systems are different enough the parentheses aren't necessary.

I am also wondering if glossing names like this actually makes sense across entirely different writing systems.  To an English speaker, the Italian name FIrenze is less familiar than the anglicized name Florence, but still readable and pronounceable since Italian and English use essentially the same alphabet.  On the other hand, دبي is completely unintelligible to the same English only speaker, while Dubai is a name they can comprehend.  I imagine the same would be true for an Arabic speaker who is unfamiliar with the Roman alphabet.

Here's an example of a Nat Geo map of Quatar showing the capitol city as "Ad Dowhah (Doha)", but no Arabic script is used.
![Screen Shot 2022-11-29 at 9 05 35 PM](https://user-images.githubusercontent.com/281482/204690203-f0e463b6-d957-49a5-8780-3fd989e09591.png)

On the other hand, Google maps shows both writing systems in the Middle East, but just one name per city in Italy.
![Screen Shot 2022-11-29 at 9 14 52 PM](https://user-images.githubusercontent.com/281482/204691421-03a8241b-1ff2-47e1-a386-ef302b459f70.png)  ![Screen Shot 2022-11-29 at 9 20 16 PM](https://user-images.githubusercontent.com/281482/204691761-fc6ebdd9-41a4-4abc-8129-a3c3c8875f63.png)

Reply:
> this example is with two languages that both use the Roman alphabet. Maybe when the writing systems are different enough the parentheses aren't necessary.

I agree that a difference in writing direction or writing systems that look very different would obviate the need for punctuation. However, it’s pretty difficult to determine the writing system of a string using expressions alone. The Mapbox Streets source includes a [`name_script`](https://docs.mapbox.com/data/tilesets/reference/mapbox-streets-v8/#name_script-text) field indicating the dominant writing system in the `name` field. That would make it straightforward to omit punctuation in cases where the two writing systems look sufficiently different.

> I am also wondering if glossing names like this actually makes sense across entirely different writing systems. To an English speaker, the Italian name Firenze is less familiar than the anglicized name Florence, but still readable and pronounceable since Italian and English use essentially the same alphabet. On the other hand, دبي is completely unintelligible to the same English only speaker, while Dubai is a name they can comprehend. I imagine the same would be true for an Arabic speaker who is unfamiliar with the Roman alphabet.

I would’ve preferred to gloss a Latin transliteration of the local name, which would’ve been more useful to most users of this style. Looking at a name in Arabic script, a native English speaker can only visually recognize it as Arabic and intuit that the local language in the area is Arabic. I think that’s better than nothing: if we omit the gloss, then it looks like the only remaining name is the local name. But if we were able to show a transliteration, then the user would be able to sound out the local name, just as with any local name in a Latin alphabet.

Unfortunately, I don’t think we have a suitable source of transliterations at the moment, apart from when the preferred language’s Wikidata label happens to be a transliteration. `name:latin` would be ideally suited for glossing a transliteration, but it currently relies on a language-independent transform that’s primarily intended for sorting and searching as opposed to display: https://github.com/onthegomap/planetiler/issues/86#issuecomment-1050039099. Meanwhile, `int_name` is [“not necessarily” in English](https://wiki.openstreetmap.org/wiki/Special:PermanentLink/2427884#Variants) but in practice is usually in English and is rarely a transliteration.

> Here's an example of a Nat Geo map of Quatar showing the capitol city as "Ad Dowhah (Doha)", but no Arabic script is used.

Different publishers have different house rules about foreign place names. World atlases usually include a note on the subject somewhere near the legend, for readers used to English-only maps of the U.S. The policy might also vary over time, along with the rest of the publisher’s house style.

National Geographic’s house style favors the local name, relegating the English name to a gloss. But Rand McNally – which has been more influential in this style’s design – puts the English name first, also limiting bilingual labels to a smaller set of cities. The statement in Rand McNally’s _Quick Reference World Atlas_ (2000, p. 1) addresses their choice of labels to gloss, punctuation, treatment of multilingual cities, and glossing-over of common words in natural features (as opposed to cities):

> In this atlas a "local-name" policy generally was used for naming cities and towns and all local topographic and water features. However, for a few major cities the Anglicized name was preferred and the local name given in parentheses, for instance, Moscow (Moskva), Vienna (Wien), Prague (Praha). In countries where more than one official language is used, a name is in the dominant local language. The generic parts of local names for topographic and water features are self-explanatory in many cases because of the associated map symbols or type styles.

Online map publishers also have policies on foreign place names, but the details only come out in drips whenever there’s a geopolitical dispute. There are public guidelines for [names on OpenStreetMap](https://wiki.openstreetmap.org/wiki/Names#Localization) and [labels in Wikidata](https://www.wikidata.org/wiki/Special:MyLanguage/Help:Label).[^wikidata] However, these data sources are only concerned with the name in a particular language, which doesn’t guide us when trying to provide more than just the local name or just a localized name.

[^wikidata]: Wikidata labels are largely based on Wikipedia article titles, with a view towards removing the disambiguating suffixes that Wikipedia adds to keep article titles unique. In turn, the English Wikipedia largely uses anglicized names, but [editors disagree](http://notes.1ec5.org/archives/2021/05/01/diacritics.html) on where to draw the line about systematic or obscure anglicizations.

Reply:
> it’s pretty difficult to determine the writing system of a string using expressions alone.  The Mapbox Streets source includes a [name_script](https://docs.mapbox.com/data/tilesets/reference/mapbox-streets-v8/#name_script-text) field

Makes sense.  Clearly we'd need OpenMapTiles to add a similar field to do anything different based on writing system.

> I would’ve preferred to gloss a Latin transliteration of the local name

Yes, that seems to be the usual style I've seen on most maps.  Google maps is the first I've seen showing multiple writing systems at once, but maybe there are others.  I guess the thing is that when the alphabet differs between two names, it's hard to tell if you're looking at an actual different name, or a transliteration.  For example, it seems like Dubai is a transliteration of the local name so there's no need for a gloss indicating a different local name.  However, when the writing system is different I guess there's no easy way to tell the difference between a transliteration and a truly different local name.